### ISSUE TITLE
Evaluate Gradients with different data points

### ISSUE THREAD

**Issue by Ulfgard** [keras-team/keras#12790]
Hi,

This is a feature request.

I am trying to implement an optimizer in Keras, for which i need to evaluate a gradient twice at the same point, but with a different mini-batch of data (e.g. to obtain an estimate of the noise on the gradient).

I found information on how to evaluate the gradient at different values of the parameters using tensorflows graph_replace, but so far i found no information on how to replace the mini-batch or even to obtain another one from Keras during training.

Is this possible in Keras? In short what i need is something equivalent to

	def get_updates(self, loss, params):
		grads = tf.gradients(loss, params) #current gradient evaluation
		loss = K.next_mini_batch(loss) # graph for loss with just the data replaced
		grads2 = tf.gradients(loss, params)
